orchagent/aclorch.cpp

```diff
@@ -3195,6 +3373,93 @@ void AclOrch::init(vector<TableConnector>& connectors, PortsOrch *portOrch, Mirr
     }
 
 
+    if (platform == VS_PLATFORM_SUBSTRING)
+    {
+        // For testing on VS the following values will be used.
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_RANGE_CAPABLE] = "true";
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_MIN] = "1";
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_MAX] = "7";
+        m_switchMetaDataCapabilities[TABLE_ACL_ENTRY_ATTR_META_CAPABLE] = "true";
+        m_switchMetaDataCapabilities[TABLE_ACL_ENTRY_ACTION_META_CAPABLE] = "true";
+        m_metaDataMgr.populateRange(1,7);
+    }
+    else
+    {
+        // check switch capability of Metadata attribute, action and range.
+        // SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE support and range values.
+        // SAI_ACL_ENTRY_ATTR_ACTION_SET_ACL_META_DATA
+        // SAI_ACL_ENTRY_ATTR_FIELD_ACL_USER_META
+
+        sai_status_t status = SAI_STATUS_SUCCESS;
+        sai_attr_capability_t capability;
+        uint16_t metadataMin = 0;
+        uint16_t metadataMax = 0;
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_MIN] = "0";
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_MAX] = "0";
+        m_switchMetaDataCapabilities[TABLE_ACL_USER_META_DATA_RANGE_CAPABLE] = "false";
+        m_switchMetaDataCapabilities[TABLE_ACL_ENTRY_ATTR_META_CAPABLE] = "false";
+        m_switchMetaDataCapabilities[TABLE_ACL_ENTRY_ACTION_META_CAPABLE] = "false";
+
+        status = sai_query_attribute_capability(gSwitchId, SAI_OBJECT_TYPE_SWITCH, SAI_SWITCH_ATTR_ACL_USER_META_DATA_RANGE, &capability);
```

Comment:
SImilar to validating USER_META_DATA capability, do we need to check ACL action set_dscp capability?

Reply:
that is already present on all platforms as we have tables using this capability.